src/Npgsql/NpgsqlDatabaseInfo.cs

```diff
@@ -180,6 +182,12 @@ internal void ProcessTypes()
         /// <returns></returns>
         protected abstract IEnumerable<PostgresType> GetTypes();
 
+        /// <summary>
+        /// Adapts the type mappings for this database.
+        /// </summary>
+        /// <param name="mappings">The mappings that are about the be bound.</param>
+        protected internal virtual void AdaptTypeMappings(IDictionary<string, NpgsqlTypeMapping> mappings) { }
```

Comment:
I don't think that it's a good idea to adopt mappings each time. It will take a lot of time to connect to CrateDB. So, it would be much better to initialize mappings using the same way for all databases. I mean that there must be a virtual method or an abstract which will populate mappings even for PostgreSQL.

Reply:
Hi @YohDeadfall the reason we are doing it this way was the problems which were discussed in following issues 
https://github.com/npgsql/npgsql/pull/2202#issuecomment-461985016
https://github.com/crate/crate-npgsql/issues/7#issuecomment-389944734

Reply:
I understand it and there is nothing to do from CrateDB side, so it's our problem (:

@roji Is it a blocker? Should it be fixed or go ahead and merge for now?

Reply:
I haven't managed to understand the reason for this "adapting" phase - I posted [this comment](https://github.com/npgsql/npgsql/pull/2202#issuecomment-441374144) a while ago but didn't get an answer... In a nutshell and as I wrote in that comment, you shouldn't be seeing any exceptions due to non-existing types, since the `externalCall` flag passed to `ConnectorTypeMapper.BindType()` should be false. An exception is only meant to be thrown when the user attempts to map an enum or composite.

So I prefer not to go ahead with this specific change unless we understand that there's a real problem, in which case I'd like to fix it rather than an additional API for removing stuff.